DifferentiationInterface/ext/DifferentiationInterfaceFiniteDiffExt/onearg.jl

```diff
@@ -56,87 +78,118 @@ function DI.prepare_derivative(
         df = similar(y)
         cache = GradientCache(df, x, fdtype(backend), eltype(y), FUNCTION_NOT_INPLACE)
     end
-    return FiniteDiffOneArgDerivativePrep(cache)
+    relstep = if isnothing(backend.relstep)
+        default_relstep(fdtype(backend), eltype(x))
+    else
+        backend.relstep
+    end
+    absstep = isnothing(backend.absstep) ? relstep : backend.relstep
+    return FiniteDiffOneArgDerivativePrep(cache, relstep, absstep)
 end
 
 ### Scalar to scalar
 
 function DI.derivative(
     f,
-    ::FiniteDiffOneArgDerivativePrep{Nothing},
+    ::FiniteDiffOneArgDerivativePrep{Nothing,Nothing,Nothing},
```

Comment:
this dispatch is not possible because you make sure in the preparation that the steps will not be `nothing`
```suggestion
    ::FiniteDiffOneArgDerivativePrep{Nothing},
```

Reply:
it also explains the error you encountered in CI, where the dispatch chain was broken and DI mistook the `prep` object in `derivative(f, prep, backend, x)` for a `y` object in `derivative(f!, y, backend, x)`